### PR TITLE
Docs: Add data links shared content

### PR DESCRIPTION
**Pull request by imatwawana**
This PR adds a shared file with data link options for visualizations and adds that file to relevant visualizations.

This PR:

- Adds shared file for data links content
- Adds shared file to appropriate visualizations
- Fixes version interpolation syntax
- Updates links for grafana-cloud

**TO DO**

- [x] Set content for canvas
- [x] Decide on content format for xy chart

### CODE REVIEW

**Comment by imatwawana** [comment]
@jdbaldry - can you just check my relative URLs please?

File: docs/sources/shared/visualizations/datalink-options.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+title: Data link options
+comments: |
+  This file is used in the following visualizations: bar chart, bar gauge, candlestick, canvas*, gauge, geomap, heatmap, histogram, stat, state timeline, status history, table, time series, trend, xy chart*
```
Comment (imatwawana) [inline docs/sources/shared/visualizations/datalink-options.md:4]:
```suggestion
  This file is used in the following visualizations: bar chart, bar gauge, candlestick, gauge, geomap, heatmap, histogram, stat, state timeline, status history, table, time series, trend, xy chart
```

File: docs/sources/panels-visualizations/visualizations/bar-chart/index.md [hunk 2]
```diff
@@ -199,13 +199,20 @@ You can set standard min/max options to define hard limits of the Y-axis. For mo
 
 {{< docs/shared lookup="visualizations/multiple-y-axes.md" source="grafana" version="<GRAFANA_VERSION>" leveloffset="+2" >}}
 
+## Data links
+
+{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" leveloffset="+2" >}}
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/bar-chart/index.md:204]:
```suggestion
{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>"  >}}
```

File: docs/sources/panels-visualizations/visualizations/bar-gauge/index.md [hunk 3]
```diff
@@ -130,7 +130,11 @@ Automatically show y-axis scrollbar when there's a large amount of data.
 This option only applies when bar size is set to manual.
 {{% /admonition %}}
 
+## Data links
+
+{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" leveloffset="+2" >}}
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/bar-gauge/index.md:135]:
```suggestion
{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" >}}
```

File: docs/sources/panels-visualizations/visualizations/candlestick/index.md [hunk 4]
```diff
@@ -110,15 +110,19 @@ If your data can't be mapped to these dimensions for some reason (for example, b
...
 
 The candlestick visualization is based on the time series visualization. It can visualize additional data dimensions beyond open, high, low, close, and volume The **Include** and **Ignore** options allow it to visualize other included data such as simple moving averages, Bollinger bands and more, using the same styles and configurations available in the [time series][time series visualization] visualization.
 
 ## Tooltip options
 
 {{< docs/shared lookup="visualizations/tooltip-options-2.md" source="grafana" version="<GRAFANA_VERSION>" >}}
 
+## Data links
+
+{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" leveloffset="+2" >}}
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/candlestick/index.md:123]:
```suggestion
{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" >}}
```

File: docs/sources/panels-visualizations/visualizations/gauge/index.md [hunk 5]
```diff
@@ -105,7 +105,11 @@ Adjust the sizes of the gauge text.
 - **Title -** Enter a numeric value for the gauge title size.
 - **Value -** Enter a numeric value for the gauge value size.
 
+## Data links
+
+{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" leveloffset="+2" >}}
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/gauge/index.md:110]:
```suggestion
{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" >}}
```

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 6]
```diff
@@ -603,10 +603,14 @@ Displays debug information in the upper right corner. This can be useful for deb
 - **None** displays tooltips only when a data point is clicked.
 - **Details** displays tooltips when a mouse pointer hovers over a data point.
 
+## Data links
+
+{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" leveloffset="+2" >}}
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:608]:
```suggestion
{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" >}}
```

File: docs/sources/panels-visualizations/visualizations/heatmap/index.md [hunk 7]
```diff
@@ -174,11 +174,15 @@ When you set the **Tooltip mode** to **Single**, this option is displayed. This
 
 Choose whether you want to display the heatmap legend on the visualization.
 
+### Data links
+
+{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" leveloffset="+3" >}}
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/heatmap/index.md:179]:
```suggestion
{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" >}}
```

File: docs/sources/panels-visualizations/visualizations/histogram/index.md [hunk 8]
```diff
@@ -121,16 +121,20 @@ Transparency of the gradient is calculated based on the values on the Y-axis. Th
...
-{{< docs/shared lookup="visualizations/legend-mode.md" source="grafana" version="<GRAFANA VERSION>" >}}
+{{< docs/shared lookup="visualizations/legend-mode.md" source="grafana" version="<GRAFANA_VERSION>" >}}
 
 ### Legend calculations
 
 Choose a [standard calculations][] to show in the legend. You can select more than one.
 
+## Data links
+
+{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" leveloffset="+2" >}}
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/histogram/index.md:132]:
```suggestion
{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" >}}
```

File: docs/sources/panels-visualizations/visualizations/stat/index.md [hunk 9]
```diff
@@ -189,7 +189,11 @@ Adjust the sizes of the gauge text.
 - **Title -** Enter a numeric value for the gauge title size.
 - **Value -** Enter a numeric value for the gauge value size.
 
+## Data links
+
+{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" leveloffset="+2" >}}
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/stat/index.md:194]:
```suggestion
{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" >}}
```

File: docs/sources/panels-visualizations/visualizations/state-timeline/index.md [hunk 10]
```diff
@@ -134,16 +134,20 @@ The visualization can be used with time series data as well. In this case, the t
...
-{{< docs/shared lookup="visualizations/legend-mode.md" source="grafana" version="<GRAFANA VERSION>" >}}
+{{< docs/shared lookup="visualizations/legend-mode.md" source="grafana" version="<GRAFANA_VERSION>" >}}
 
 ## Tooltip options
 
 {{< docs/shared lookup="visualizations/tooltip-options-1.md" source="grafana" version="<GRAFANA_VERSION>" >}}
 
+## Data links
+
+{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" leveloffset="+2" >}}
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/state-timeline/index.md:145]:
```suggestion
{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" >}}
```

File: docs/sources/panels-visualizations/visualizations/status-history/index.md [hunk 11]
```diff
@@ -114,16 +114,20 @@ To assign colors to boolean or string values, use the [Value mappings][].
...
-{{< docs/shared lookup="visualizations/legend-mode.md" source="grafana" version="<GRAFANA VERSION>" >}}
+{{< docs/shared lookup="visualizations/legend-mode.md" source="grafana" version="<GRAFANA_VERSION>" >}}
 
 ## Tooltip options
 
 {{< docs/shared lookup="visualizations/tooltip-options-1.md" source="grafana" version="<GRAFANA_VERSION>" >}}
 
+## Data links
+
+{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" leveloffset="+2" >}}
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/visualizations/status-history/index.md:125]:
```suggestion
{{< docs/shared lookup="visualizations/datalink-options.md" source="grafana" version="<GRAFANA_VERSION>" >}}
```

File: docs/sources/shared/visualizations/datalink-options.md [hunk 12]
```diff
@@ -0,0 +1,15 @@
+---
+title: Data link options
+comments: |
+  This file is used in the following visualizations: bar chart, bar gauge, candlestick, gauge, geomap, heatmap, histogram, stat, state timeline, status history, table, time series, trend, xy chart
+---
+
+Data links allow you to provide more granular context to your links. You can create links that include the series name or even the value under the cursor.
+
+For each data link, you'll set the following options:
```
Comment (imatwawana) [inline docs/sources/shared/visualizations/datalink-options.md:9]:
```suggestion
For each data link, set the following options:
```

File: docs/sources/shared/visualizations/datalink-options.md [hunk 13]
```diff
@@ -0,0 +1,15 @@
+---
+title: Data link options
+comments: |
+  This file is used in the following visualizations: bar chart, bar gauge, candlestick, gauge, geomap, heatmap, histogram, stat, state timeline, status history, table, time series, trend, xy chart
+---
```
Comment (nmarrs) [inline docs/sources/shared/visualizations/datalink-options.md:5]: ❤️

File: docs/sources/shared/visualizations/datalink-options.md [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+
+Data links allow you to provide more granular context to your links. You can create links that include the series name or even the value under the cursor.
```
Comment (nmarrs) [inline docs/sources/shared/visualizations/datalink-options.md:7]: I think we may want to rephrase this, I think the key value add here maybe is more around adding the ability to link to external resources / other dashboards and include contextual dashboard data / metadata - this needs some work but I think more of the general idea 😅

**Review by nmarrs — APPROVED** [review]
Overall LGTM! I think we might want to consider rewording definition of data links / provide a few concrete examples of what they can be used for

File: docs/sources/shared/visualizations/datalink-options.md (hunk 14 above)
Comment (imatwawana) [reply docs/sources/shared/visualizations/datalink-options.md:7]: I took this from the main Configure data links page, so we should probably update both now vs letting this go. Let me circle back with a proposed update.

**Comment by imatwawana** [comment]
> @jdbaldry - can you just check my relative URLs please?

Nvm! I've tested them locally in grafana and cloud and they're working.

**Review by baldm0mma — APPROVED** [review]

**Review by jdbaldry — APPROVED** [review]
nice, thanks

File: docs/sources/panels-visualizations/configure-data-links/index.md [hunk 15]
```diff
@@ -26,7 +26,7 @@ weight: 80
 
 # Configure data links
 
-Data links allow you to provide more granular context to your links. You can create links that include the series name or even the value under the cursor. For example, if your visualization shows four servers, you can add a data link to one or two of them. You can also link panels using data links.
+Data links allow you to link to other panels, dashboards, and external resources while maintaining the context of the source panel. You can create links that include the series name or even the value under the cursor. For example, if your visualization shows four servers, you can add a data link to one or two of them. You can also link panels using data links.
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/configure-data-links/index.md:29]: @nmarrs - What about this? I've used this again [here](https://github.com/grafana/grafana/pull/86893/files#diff-c9ac8cb4f306982a9a04badfad45725245a480cc08e8fd7a0ef8def27418c914R7).

Comment (nmarrs) [reply docs/sources/panels-visualizations/configure-data-links/index.md:29]: Looks good! :shipit:

Comment (imatwawana) [inline docs/sources/panels-visualizations/configure-data-links/index.md:29]: Removing redundant sentence
```suggestion
Data links allow you to link to other panels, dashboards, and external resources while maintaining the context of the source panel. You can create links that include the series name or even the value under the cursor. For example, if your visualization shows four servers, you can add a data link to one or two of them.
```

**Pull request merged**